src/controllers/colorjsproxy.cpp

```diff
@@ -1,20 +1,26 @@
 #include "controllers/colorjsproxy.h"
+#include "controllers/controllerengine.h"
 
 ColorJSProxy::ColorJSProxy(QScriptEngine* pScriptEngine)
-    : m_pScriptEngine(pScriptEngine),
-      m_predefinedColorsList(makePredefinedColorsList(pScriptEngine)){};
+        : m_pScriptEngine(pScriptEngine),
+          m_predefinedColorsList(makePredefinedColorsList()) {
+}
 
-ColorJSProxy::~ColorJSProxy() {};
+ColorJSProxy::~ColorJSProxy() {
+}
 
 QScriptValue ColorJSProxy::predefinedColorFromId(int iId) {
-    PredefinedColorPointer color(Color::kPredefinedColorsSet.predefinedColorFromId(iId));
-    return jsColorFrom(color);
-};
+    return m_predefinedColorsList.property(iId);
+}
 
 Q_INVOKABLE QScriptValue ColorJSProxy::predefinedColorsList() {
     return m_predefinedColorsList;
 }
 
+void ColorJSProxy::setDefaultColor(const QColor& defaultColor) {
+    m_predefinedColorsList = makePredefinedColorsList(defaultColor);
+}
+
 QScriptValue ColorJSProxy::jsColorFrom(PredefinedColorPointer predefinedColor) {
```

Comment:
This function does not store the pointer, so it should take a const reference. 
This is not part of this PR, but the change suits to simplify the using code.